csvqb/csvqb/models/rdf/newunitresource.py

```diff
@@ -0,0 +1,50 @@
+"""
+New Unit
+--------
+"""
+from typing import Annotated, Optional
+
+from rdflib import RDFS, XSD
+from sharedmodels.rdf.namespaces import OM2, QUDT
+from sharedmodels.rdf import (
+    NewMetadataResource,
+    Triple,
+    PropertyStatus,
+    map_resource_to_uri,
+    map_to_literal_with_datatype,
+    ExistingResource,
+    MaybeResource,
+)
+
+
+class NewUnitResource(NewMetadataResource):
+    """
+    New RDF Resource representing a unit.
+
+    Uses the OM2 & QUDT ontologies to define relationship with existing parent unit.
+    """
+
+    source_uri: Annotated[
+        Optional[ExistingResource],
+        Triple(RDFS.isDefinedBy, PropertyStatus.recommended, map_resource_to_uri),
+    ]
+
+    base_unit_uri: Annotated[
+        MaybeResource["NewUnitResource"],
+        Triple(OM2.hasUnit, PropertyStatus.optional, map_resource_to_uri),
```

Comment:
`NewUnitResource` extending `NewMetadataResouce` - We could't understand what the triple class is doing and why we hadn't used it on other classes which extend `NewMetadataResource`